sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java

```diff
@@ -170,12 +204,13 @@ public final CalendarInterval getInterval(int ordinal) {
 
     @Override
     public final InternalRow getStruct(int ordinal, int numFields) {
-      throw new NotImplementedException();
+      return ColumnVectorUtils.toRow(parent.column(ordinal).getStruct(rowId));
     }
 
     @Override
     public final ArrayData getArray(int ordinal) {
-      throw new NotImplementedException();
+      ColumnVector.Array array = parent.column(ordinal).getArray(rowId);
+      return ColumnVectorUtils.toGenericArray(array);
```

Comment:
I think for this and getStruct, it will be too slow. We are materializing the collection which we should avoid if possible. 

I think we need to clean up InternalRow's interface and then have getStruct return that.